src/Table/TableBody.js

```diff
@@ -144,22 +139,12 @@ const TableBody = React.createClass({
 
   getInitialState() {
     return {
-      muiTheme: this.context.muiTheme || getMuiTheme(),
       selectedRows: this._calculatePreselectedRows(this.props),
     };
   },
 
-  getChildContext() {
-    return {
-      muiTheme: this.state.muiTheme,
-    };
-  },
-
-  componentWillReceiveProps(nextProps, nextContext) {
-    const newState = {
-      muiTheme: nextContext.muiTheme || this.state.muiTheme,
-    };
-
+  componentWillReceiveProps(nextProps) {
+    const newState = {};
```

Comment:
It gets called in either case, since `newState.selectedRows = this._calculatePreselectedRows(nextProps);` is unconditional. That looks inefficient, but I'm not going to chase that down here. :smile: